src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,133 @@
+package duke.parser;
+
+import duke.Duke;
+import duke.command.*;
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+import java.time.LocalDate;
+
+public class Parser {
+    public static final String COMMAND_BYE = "bye";
+    public static final String COMMAND_LIST = "list";
+    public static final String COMMAND_TODO = "todo";
+    public static final String COMMAND_DEADLINE = "deadline";
+    public static final String COMMAND_EVENT = "event";
+    public static final String COMMAND_DONE = "done";
+    public static final String COMMAND_DELETE = "delete";
+    public static final String COMMAND_FIND = "find";
+    public static final String COMMAND_HELP = "help";
+    public static final String COMMAND_OCCUR = "occur";
+
+    private static String getCommand(String fullCommand) {
+        return fullCommand.split(" ")[0];
+    }
+
+    private static String getTaskDescription(String fullCommand) throws DukeException {
+        String command = getCommand(fullCommand);
+
+        if (fullCommand.split(command).length < 2
+            || fullCommand.split(command).equals(" ")) {
+            throw new DukeException("☹ OOPS!!! Missing duke.task description!");
+        } else {
+            switch (command) {
+            case COMMAND_TODO:
+                return fullCommand.substring(fullCommand.indexOf(' ') + 1);
+            case COMMAND_DEADLINE:
+                if (!fullCommand.contains("/by ")) {
+                    throw new DukeException("☹ OOPS!!! Missing or incorrect /by statement");
+                }
+                return fullCommand.substring(fullCommand.indexOf(' ') + 1, fullCommand.indexOf("/by ") - 1);
+            case COMMAND_EVENT:
+                if (!fullCommand.contains("/at ")) {
+                    throw new DukeException("☹ OOPS!!! Missing or incorrect /at statement");
+                }
+                return fullCommand.substring(fullCommand.indexOf(' ') + 1, fullCommand.indexOf("/at ") - 1);
+            default:
+                return null;
+            }
+        }
+    }
+
+    private static LocalDate getTaskDate(String fullCommand) throws DukeException {
+        String taskDateString;
+        switch (getCommand(fullCommand)) {
+        case COMMAND_DEADLINE:
+            taskDateString = fullCommand.split("/by ")[1];
```

Comment:
Perhaps for the magic literal (index 1), can make it into a constant with a descriptive name to increase readability.